.github/workflows/main.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish Home Assistant Docker images
+
+on:
+  push
+
+jobs:
+  publish:
+    name: Build and publish image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Publish
+        uses: home-assistant/builder@master
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          IMAGE_NAME: hassio-to-pubsub
+          SOURCE_DIRECTORY: hassio-to-pubsub-source
```

Comment:
I was under the impression we would have the add-on simply pull the image we build from the source. I guess it would be more appropriate to only have one that we can use as the add-on, but not too sure on how we would approach that. 

Isn't the idea that the source is built into the image (because the PI is too slow) and the add-on uses the image in its `Dockerfile`?

Reply:
Having a Dockerfile with a single `FROM` statement pointing to our pre-built image was just a workaround. We should be able to have GitHub Actions build/push a single image to Docker Hub, then specify that image in `config.json` via the `image` key, which means the Pi doesn't have to build anything at all.

Reply:
If we specify the image in `config.json` for the `-source`, wouldn't the add-on try and use the Dockerfile too when trying to be installed/run?

Reply:
I don't think so. If `image` is specified, Supervisor should download that rather than building.

I think the best thing to do would be to take a look at some official/popular add-ons and see how they've structured their builds.